images/kyverno/main.tf

```diff
@@ -0,0 +1,62 @@
+terraform {
+  required_providers {
+    apko = { source = "chainguard-dev/apko" }
+  }
+}
+
+locals {
+  components = toset(["admission", "background", "cleanup", "cli", "init", "reports"])
+
+  packages = merge({
+    for k, v in local.components : k => "kyverno-${k}-controller"
+    }, {
+    "admission" : "kyverno",
+    "cli" : "kyverno-cli",
+    "init" : "kyverno-init-container",
+  })
+
+  repositories = merge({
+    for k, v in local.components : k => "${var.target_repository}-${k}-controller"
+    }, {
+    "admission" : var.target_repository,
+    "cli" : "${var.target_repository}-cli",
+    "init" : "${var.target_repository}pre",
+  })
+}
+
+variable "target_repository" {
+  description = "The docker repo into which the image and attestations should be published."
+}
+
+module "latest" {
+  for_each = local.components
+  source   = "../../tflib/publisher"
+
+  target_repository = local.repositories[each.key]
+  config            = file("${path.module}/configs/latest.${each.key}.apko.yaml")
+}
+
+module "version-tags" {
+  for_each = local.components
+  source   = "../../tflib/version-tags"
+
+  package = local.packages[each.key]
+  config  = module.latest[each.key].config
+}
+
+module "test-latest" {
```

Comment:
No dev variants?

Reply:
it doesn't have them currently, so I'm not adding them